lib/ansible/modules/storage/netapp/sf_snapshot_schedule_manager.py

```diff
@@ -0,0 +1,432 @@
+                    logger.debug("CHANGED: snapshot name needs to be updated")
+                    update_schedule = True
+                    changed = True
+
+                elif self.pause is not None and schedule_detail.paused != self.pause:
+                    logger.debug("CHANGED: Schedule pause status needs to be updated")
+                    update_schedule = True
+                    changed = True
+
+                elif self.recurring is not None and schedule_detail.recurring != self.recurring:
+                    logger.debug("CHANGED: Recurring status needs to be updated")
+                    update_schedule = True
+                    changed = True
+
+                elif self.starting_date is not None and schedule_detail.starting_date != self.starting_date:
+                    logger.debug("CHANGED: Starting date needs to be updated")
+                    update_schedule = True
+                    changed = True
+
+                elif self.time_interval_minutes is not None or self.time_interval_hours is not None \
+                        or self.time_interval_days is not None:
+
+                    temp_frequency = netapp_utils.TimeIntervalFrequency(days=self.time_interval_days,
+                                                                        hours=self.time_interval_hours,
+                                                                        minutes=self.time_interval_minutes)
+
+                    if schedule_detail.frequency.days != temp_frequency.days or \
+                            schedule_detail.frequency.hours != temp_frequency.hours \
+                            or schedule_detail.frequency.minutes != temp_frequency.minutes:
+                        logger.debug("CHANGED: schedule interval needs to be updated")
+                        update_schedule = True
+                        changed = True
+
+        else:
+            if self.state == 'present':
+                logger.debug(
+                    "CHANGED: schedule does not exist, but requested state is "
+                    "'present'")
+                changed = True
+
+        if changed:
+            if self.module.check_mode:
+                logger.debug('skipping changes due to check mode')
+            else:
+                if self.state == 'present':
+                    if not schedule_exists:
+                        self.create_schedule()
+                    elif update_schedule:
+                        self.update_schedule()
+
+                elif self.state == 'absent':
+                    self.delete_schedule()
+        else:
+            logger.debug("exiting with no changes")
+
+        self.module.exit_json(changed=changed)
+
+
+def main():
+    v = SolidFireSnapShotSchedule()
+
+    try:
+        v.apply()
+    except:
+        err = get_exception()
+        logger.debug("Exception in apply(): \n%s" % format_exc(err))
+        raise
```

Comment:
ending module execution in  an exception is a bug

use fail_json, if you want to pass back traceback information use the 'exception' key in the returned json